src/pydocstyle/checker.py

```diff
@@ -486,8 +501,8 @@ def _check_section_underline(cls, section_name, context, indentation):
 
         if not dash_line_found:
             yield violations.D407(section_name)
-            if next_non_empty_line_offset == 0:
-                yield violations.D410(section_name)
+            if next_non_empty_line_offset > 0:
```

Comment:
`next_non_empty_line_offset` is a confusing name. How about something like `blank_lines_after_header`?